### PR TITLE
Optim-wip: Add descriptions and argument documentation to losses

### PR DESCRIPTION
**Pull request by greentfrapp**

### CODE REVIEW

File: captum/optim/_core/loss.py [hunk 1]
```diff
@@ -189,6 +189,14 @@ def wrapper(*args, **kwargs) -> object:
 class LayerActivation(BaseLoss):
     """
     Maximize activations at the target layer.
+    This is the most basic loss available and it simply returns the activations in
+    their original form.
+
+    Args:
+        target (nn.Module):  The layer to optimize for.
+        batch_index (int, optional):  The index of the image to optimize if we
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:197]: The `batch_index` documentation is missing `Default: None` below the description for all of the batch_index docs I think.

File: captum/optim/_core/loss.py [hunk 2]
```diff
@@ -224,6 +241,26 @@ def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
...
+
+    Args:
+        target (nn.Module):  The layer to containing the channel to optimize for.
+        channel_index (int):  The index of the channel to optimize for.
+        x (int, optional):  The x coordinate of the neuron to optimize for. If
+            unspecified, defaults to center, or one unit left of center for even
+            lengths.
+        y (int, optional):  The y coordinate of the neuron to optimize for. If
+            unspecified, defaults to center, or one unit up of center for even
+            heights.
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:258]: These docs are also missing `Default: None`.

File: captum/optim/_core/loss.py [hunk 3]
```diff
@@ -464,6 +575,23 @@ class NeuronDirection(BaseLoss):
...
+
+    Args:
+        target (nn.Module):  The layer to optimize for.
+        vec (torch.Tensor):  Vector representing direction to align to.
+        x (int, optional):  The x coordinate of the neuron to optimize for. If
+            unspecified, defaults to center, or one unit left of center for even
+            lengths.
+        y (int, optional):  The y coordinate of the neuron to optimize for. If
+            unspecified, defaults to center, or one unit up of center for even
+            heights.
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:589]: These docs are missing `Default: None`.

File: captum/optim/_core/loss.py [hunk 4]
```diff
@@ -542,6 +679,23 @@ def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
...
+        wx (int, optional):  Length of neurons to apply the weights to, along the
+            x-axis.
+        wy (int, optional):  Length of neurons to apply the weights to, along the
+            y-axis.
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:698]: These docs are all missing `Default: None`

File: captum/optim/_core/loss.py [hunk 5]
```diff
@@ -201,6 +209,15 @@ def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
...
     Maximize activations at the target layer and target channel.
+    This loss maximizes the activations of a target channel in a specified target
+    layer, and can be useful to determine what features the channel is excited by.
+
+    Args:
+        target (nn.Module):  The layer to containing the channel to optimize for.
+        channel_index (int):  The index of the channel to optimize for.
+        batch_index (int, optional):  The index of the image to optimize if we
+            optimizing a batch of images. If unspecified, defaults to all images
+            in the batch.
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:220]: I'm not sure if we should be calling the activations `images` in the batch index docs, as the activations can be images, or 2D / 4D activations.

Comment (ProGamerGov) [reply captum/optim/_core/loss.py:220]: There's also a grammatical error in the batch_index docs: "optimize if we optimizing a batch of", should be "optimize if we **are** optimizing a batch of".

File: captum/optim/_core/loss.py [hunk 6]
```diff
@@ -307,6 +371,15 @@ def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
 class L2(BaseLoss):
     """
     L2 norm of the target layer, generally used as a penalty.
+
+    Args:
+        target (nn.Module):  The layer to optimize for.
+        constant (float):  Constant threshold to deduct from the activations.
+            Defaults to 0.
+        epsilon (float):  Small value to add to L2 prior to sqrt. Defaults to 1e-6.
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:379]: I think the documentation format requires `Default: <default_value>` on a separate line.

File: captum/optim/_core/loss.py [hunk 7]
```diff
@@ -307,6 +371,15 @@ def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
 class L2(BaseLoss):
     """
     L2 norm of the target layer, generally used as a penalty.
+
+    Args:
+        target (nn.Module):  The layer to optimize for.
+        constant (float):  Constant threshold to deduct from the activations.
```
Comment (ProGamerGov) [inline captum/optim/_core/loss.py:377]: The doc should be listed as optional, as there is a default value: `constant (float, optional)`

File: captum/optim/_core/loss.py (hunk 6 above)
Comment (ProGamerGov) [reply captum/optim/_core/loss.py:379]: The epsilon doc is also missing the `optional` word

**Comment by NarineK** [comment]
@ProGamerGov, @greentfrapp, I'm thinking to merge this first since it only contains docs and I don't see any merge conflicts.

**Comment by ProGamerGov** [comment]
@NarineK Yeah, lets do that!
